### PR TITLE
[ISSUE-849][REFACTOR] Migrate master and common Celeborn Configuration System

### LINKED ISSUE #849: [REFACTOR] Migrate master and common Celeborn Configuration System
### Describe the intention you'd like
A clear and concise description of what you want to do.

/cc @who-need-to-know

/assign @who-can-help-you

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related issues.
close #849

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1378,17 +1388,47 @@ object CelebornConf extends Logging {
...
+      .internal
+      .doubleConf
+      .createWithDefault(0.1)
+
+  val MASTER_OFFER_SLOTS_EXTRA_SIZE: ConfigEntry[Int] =
+    buildConf("celeborn.master.offerSlots.extraSize")
+      .withAlternative("rss.offer.slots.extra.size")
+      .categories("master")
+      .version("0.2.0")
+      .doc("")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1418]: ping @waitinfuture What should I write here?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -1378,17 +1388,47 @@ object CelebornConf extends Logging {
...
+      .intConf
+      .createWithDefault(2)
+
+  val MASTER_OFFER_SLOTS_ALGORITM: ConfigEntry[String] =
+    buildConf("celeborn.master.offerSlots.algorithm")
+      .withAlternative("rss.offer.slots.algorithm")
+      .categories("master")
+      .version("0.2.0")
+      .doc("")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1427]: ditto

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 3]
```diff
@@ -1527,9 +1567,23 @@ object CelebornConf extends Logging {
...
+    buildConf("celeborn.port.maxRetries")
+      .withAlternative("rss.master.port.maxretry")
+      .categories("master", "worker", "client")
+      .doc("When port is occupied,we will retry for max retry times.")
+      .version("0.2.0")
+      .intConf
+      .createWithDefault(1)
+
+  val HA_CLIENT_MAX_RETRIES: ConfigEntry[Int] =
+    buildConf("celeborn.ha.client.maxRetries")
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1580]: @pan3793 @waitinfuture It's for client to connect to master endpoints

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 1 above)
Comment (waitinfuture) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1428]: As we discussed, let's use the following names
```
celeborn.slots.assign.policy
celeborn.slots.assign.loadAware.numDiskGroups
celeborn.slots.assign.loadAware.diskGroupGradient
celeborn.slots.assign.extraSlots
```

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture How about current.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 4]
```diff
@@ -1412,17 +1423,45 @@ object CelebornConf extends Logging {
...
+      .version("0.2.0")
+      .doubleConf
+      .createWithDefault(0.1)
+
+  val SLOTS_ASSIGN_EXTRA_SLOTS: ConfigEntry[Int] =
+    buildConf("celeborn.slots.assign.extraSlots")
+      .withAlternative("rss.offer.slots.extra.size")
+      .categories("master")
+      .version("0.2.0")
+      .doc("Extra slots size when master assign slots.")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1451]: size or number?

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 5]
```diff
@@ -1412,17 +1423,45 @@ object CelebornConf extends Logging {
...
+      .intConf
+      .createWithDefault(2)
+
+  val SLOTS_ASSIGN_POLICY: ConfigEntry[String] =
+    buildConf("celeborn.slots.assign.policy")
+      .withAlternative("rss.offer.slots.algorithm")
+      .categories("master")
+      .version("0.2.0")
+      .doc("Policy for master to assign slots.")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1460]: please document the options

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 6]
```diff
@@ -1412,17 +1423,45 @@ object CelebornConf extends Logging {
...
       .createWithDefault(1000)
 
-  /**
-   * @return This configuration is a guidance for load-aware slot allocation algorithm. This value
-   *         is control how many disk groups will be created.
-   */
-  def diskGroups(conf: CelebornConf): Int = {
-    conf.getInt("rss.disk.groups", 5)
-  }
+  val SLOTS_ASSIGN_LOAD_AWARE_DISK_GROUP_NUM: ConfigEntry[Int] =
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1426]: SLOTS_ASSIGN_LOADAWARE_DISKGROUP_NUM

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 7]
```diff
@@ -1412,17 +1423,45 @@ object CelebornConf extends Logging {
...
+      .doc("This configuration is a guidance for load-aware slot allocation algorithm. " +
+        "This value is control how many disk groups will be created.")
+      .version("0.2.0")
+      .intConf
+      .createWithDefault(5)
 
-  def diskGroupGradient(conf: CelebornConf): Double = {
-    conf.getDouble("rss.disk.group.gradient", 0.1)
-  }
+  val SLOTS_ASSIGN_LOAD_AWARE_DISK_GROUP_GRADIENT: ConfigEntry[Double] =
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1436]: SLOTS_ASSIGN_LOADAWARE_DISKGROUP_GRADIENT

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 8]
```diff
@@ -1561,9 +1600,23 @@ object CelebornConf extends Logging {
...
   }
 
-  def masterPortMaxRetry(conf: CelebornConf): Int = {
-    conf.getInt("rss.master.port.maxretry", 1)
-  }
+  val PORT_MAX_RETRY: ConfigEntry[Int] =
+    buildConf("celeborn.port.maxRetries")
+      .withAlternative("rss.master.port.maxretry")
+      .categories("master", "worker", "client")
+      .doc("When port is occupied,we will retry for max retry times.")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1607]: nit: space after `,`

**Comment by AngersZhuuuu** [comment]
all done

**Review by pan3793 — APPROVED** [review]

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**